decidim-core/app/queries/decidim/metric_count.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This query counts registered users from a collection of organizations
+  # in an optional interval of time.
+  class MetricCount < Rectify::Query
+    def self.for(organization, metric, counter_type: :count, counter_field: :cumulative, group_by: :day)
+      new(organization, metric, counter_type: counter_type, counter_field: counter_field, group_by: group_by).query
+    end
+
+    def initialize(organization, metric, counter_type: :count, counter_field: :cumulative, group_by: :day)
+      @organization = organization
+      @metric = metric
+      @counter_type = counter_type
+      @counter_field = counter_field
+      @group_by = group_by
+    end
+
+    def query
+      query = Decidim::Metric.where(metric_type: @metric, organization: @organization)
+      case @counter_type
+      when :count
+        query.group(@group_by).sum(@counter_field).max.try(:last) || 0
```

Comment:
First, `@group_by` is used to group together all metric registries by his value. It is not really used for area charts, but it will for linecharts and rowcharts.

`max` returns the highest value in the `@group_by` field. For example, for a day grouping, it will return the most recent metric registry.

Reply:
We tried it in the console and it always returned the key-value pair with the highest key, instead of the highest value. The `group_by` returns a hash, so:

```
irb(main):005:0> {a: 1, b:4, z: 0}.max
=> [:z, 0]
irb(main):006:0> {a: 1, b:4, z: 0}.values.max
=> 4
```

I think you should use `values.max` instead of directly using `max`, unless we're missing something.